src/components/Demo.svelte

```diff
@@ -69,65 +70,106 @@ let languages: {
   },
 ] as const;
 
+let currentVersion: number = 1; // Needs to be updated when a new version is released.
+let fontSize = $state("1em");
+let simplify = $state(true);
+let flatSwitch = $state(true);
+let highlight = $state(true);
+let parsedColorList = undefined;
 const urlParams = new URLSearchParams(window.location.search);
-if (urlParams.has("go")) {
-  languageCode.Go = LZString.decompressFromEncodedURIComponent(
-    urlParams.get("go"),
-  );
-}
-if (urlParams.has("c")) {
-  languageCode.C = LZString.decompressFromEncodedURIComponent(
-    urlParams.get("c"),
-  );
-}
-if (urlParams.has("c++")) {
-  languageCode["C++"] = LZString.decompressFromEncodedURIComponent(
-    urlParams.get("c++"),
-  );
-}
-if (urlParams.has("python")) {
-  languageCode.Python = LZString.decompressFromEncodedURIComponent(
-    urlParams.get("python"),
-  );
-}
-if (urlParams.has("typescript")) {
-  languageCode.TypeScript = LZString.decompressFromEncodedURIComponent(
-    urlParams.get("typescript"),
-  );
-}
-if (urlParams.has("tsx")) {
-  languageCode.TSX = LZString.decompressFromEncodedURIComponent(
-    urlParams.get("tsx"),
+let selection = $state(
+  languages.find(
+    (lang) =>
+      lang.language.toLowerCase() === urlParams.get("language")?.toLowerCase(),
+  ) ?? languages[0],
+);
+/**
```

Comment:
That's a bit of a nitpick, but important - regular comments should start with `/*` not `/**`.

`/**` is for JSDoc/typedoc comments describing variables of functions.
Since this is just a block comment, it should use `/*`.